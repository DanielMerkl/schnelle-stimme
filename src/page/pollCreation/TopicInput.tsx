import React, { FC, useState } from 'react';
import { TextField } from '@material-ui/core';

interface Props {
  value: string;
  onChange: (topic: string) => void;
}

export const TopicInput: FC<Props> = ({ value, onChange }) => {
  const [wasTargeted, setWasTargeted] = useState(false);

  return (
    <TextField
      id="topic"
      label="Thema / Frage"
      variant="outlined"
      required
      fullWidth
      autoFocus
      error={wasTargeted && value === ''}
      value={value}
      onChange={(event) => onChange(event.target.value)}
      onBlur={() => setWasTargeted(true)}
    />
  );
};
