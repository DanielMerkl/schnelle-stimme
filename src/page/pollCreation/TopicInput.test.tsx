import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { TopicInput } from './TopicInput';

describe('TopicInput', () => {
  it('renders correctly', () => {
    const { container } = render(
      <TopicInput value={'Banana'} onChange={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });

  it('displays error state when input is empty and was targeted', async () => {
    const { rerender, getByLabelText } = render(
      <TopicInput value={''} onChange={jest.fn()} />
    );

    const topicInput = getByLabelText('Thema / Frage *');

    expect(topicInput.getAttribute('aria-invalid')).toEqual('false');

    fireEvent.blur(topicInput);

    expect(topicInput.getAttribute('aria-invalid')).toEqual('true');

    rerender(<TopicInput value={'Banane'} onChange={jest.fn()} />);

    expect(topicInput.getAttribute('aria-invalid')).toEqual('false');

    rerender(<TopicInput value={''} onChange={jest.fn()} />);

    expect(topicInput.getAttribute('aria-invalid')).toEqual('true');
  });
});
