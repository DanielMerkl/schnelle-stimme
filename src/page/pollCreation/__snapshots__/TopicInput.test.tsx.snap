// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicInput renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
      data-shrink="true"
      for="topic"
      id="topic-label"
    >
      Thema / Frage
      <span
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="topic"
        required=""
        type="text"
        value="Banana"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-61 PrivateNotchedOutline-legendNotched-62"
        >
          <span>
            Thema / Frage
             *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
